src/controllers/filter_post.rs

```diff
@@ -1,157 +1,150 @@
-    );
-    for i in 1..number_of_pages + 1 {
-        pnav.push(i.to_string())
-    }
-
-    if shared_state2.len() >= 3 {
-        for i in 0..3 {
-            plinks.push(shared_state2[i].post_title.clone());
-            pids.push(shared_state2[i].post_id);
-        }
-    } else {
-        for i in 0..shared_state2.len() {
-            plinks.push(shared_state2[i].post_title.clone());
-            pids.push(shared_state2[i].post_id);
-        }
-    }
+    if category != 0 { posts2 = get_filtered_from_database_by_category(category.clone()).await.unwrap();}
+    else{posts2 = get_filtered_from_database_by_category2().await.unwrap();}
+    //posts2 = get_filtered_from_database_by_category(category.clone()).await.unwrap();
+    if category != 0 { number_of_posts_vector = count_of_get_filtered_from_database_by_category(category).await;}
+    else{number_of_posts_vector = count_of_get_filtered_from_database_by_category2().await;}
+
+    let m2 = get_vec_len_of_count(number_of_posts_vector);
+    let number_of_pages: i64 = (m2 + 2) / global_number_of_items_per_page_64();
+    (1..number_of_pages + 1)
+        .into_iter()
+        .for_each(|i| pnav.push(i as i32));
+    posts2.iter().for_each(|post| {post_id_with_title.insert(post.post_id,post.post_title.clone());});
+    let plinks = posts2.iter().map(|post| post.post_title.clone()).collect();
+    let pids = posts2.iter().map(|post1| post1.post_id.clone()).collect();
+    println!("hashmap {:?}",post_id_with_title);
 
     let template = BlogTemplate {
-        index_id: &vec![],
-        index_title: String::from("Blogs"),
+        post_id_title:post_id_with_title,
+        category_id_title: category_id_with_title,
+        index_id: &pids,
+        index_title: String::from("Posts"),
+        page_number: &1,
+        category_name: &category_name,
         index_links: &plinks,
         index_sec: &psec,
         page_nav_links: &pnav,
         current_url_page: current_url,
     };
-
-    match template.render() {
-        Ok(html) => Html(html).into_response(),
-        Err(err) => (
+    template.render().map(|html| Html(html)).map_err(|err| {
+        (
             StatusCode::INTERNAL_SERVER_ERROR,
-            format!("Failed to render template. Error {}", err),
+            format!("Failed to render {}", err),
         )
-            .into_response(),
-    }
+    })
 }
 
-pub async fn blogs(Path(category): Path<String>) -> impl IntoResponse {
-    println!("category {} page number", category);
-    let mut psec: Vec<String> = Vec::new();
+pub async fn blogs(Path(category): Path<i32>) -> impl IntoResponse {
+    println!("category {}", category);
+    let mut pids: Vec<i32> = vec![];
+    let mut plinks: Vec<String> = vec![];
+    let mut psec: Vec<String> = vec![];
+    let temps = get_category_name_by_id(category).await;
+    let iters = temps.iter();
+    let mut category_name = "".to_string();
+    for i in iters{
+        category_name = i.category_name.clone();
+    }
```

Comment:
Why we're assigning many values to one variable ? If `temps` is mean to be a vector with one element ,we can do it like below
```rust
/*
    let iters = temps.iter();
    let mut category_name = "".to_string();
    for i in iters{
        category_name = i.category_name.clone();
    }
*/
let category_name = temps.first().unwrap_or_default();
```

Reply:
It's fixed